### PR TITLE
Scaffold entities

### PR DESCRIPTION
**Pull request by misha130**
### Pull Request summary

Adds all the entities.
Adds all the configuration.
Scaffolded from - https://github.com/mcalexster/mcalex_codidact/blob/mcalex/skeleton/Database/Scripts/base/SchemaProposalCurrent_withGenCols.sql

After scaffolding I moved all the configuration files from the DbContext file to each separate configuration. I also added base classes to all the entities so to remove typos.

Also removed all redundant property renames from the configurations.

This took me around 4 hours so its not like you can delete everything back and scaffold it again.
**This is why this is a draft.** 
Problem is the way I see it if I don't optimize it there is a lot of repeated code and redundant code. 

Whats missing - Audit tables are not automatically filled on changes though this is fairly easy to do.

### Link to Project Card/Issue

#18 

### Pull Request Tasklist

- [x] Pull request references an issue
- [x] Branch follows contributing guidelines format
- [x] Pull request targets the `develop` branch

### CODE REVIEW

File: src/Application/Members/MembersRepository.cs [hunk 1]
```diff
@@ -1,6 +1,7 @@
 ﻿using Codidact.Core.Application.Common.Interfaces;
 using Codidact.Core.Domain.Entities;
 using Microsoft.EntityFrameworkCore;
+using System;
```
Comment (asynts) [inline src/Application/Members/MembersRepository.cs]:
```suggestion
```

File: src/Infrastructure/Persistence/ApplicationDbContext.cs [hunk 2]
```diff
@@ -62,15 +98,20 @@ public override Task<int> SaveChangesAsync(CancellationToken cancellationToken =
 
         protected override void OnModelCreating(ModelBuilder modelBuilder)
         {
-            modelBuilder.ApplyConfigurationsFromAssembly(Assembly.GetExecutingAssembly());
+            {
+                modelBuilder.HasPostgresEnum("audit", "history_activity_type", new[] { "CREATE", "UPDATE_BEFORE", "UPDATE_AFTER", "DELETE" })
+                    .HasPostgresExtension("adminpack");
```
Comment (asynts) [inline src/Infrastructure/Persistence/ApplicationDbContext.cs:103]: Why do you add the "adminpack" extension? The entity framework should not directly interact with the filesystem of the database server.

File: src/Infrastructure/Infrastructure.csproj [hunk 3]
```diff
@@ -17,4 +17,8 @@
     <ProjectReference Include="..\Application\Application.csproj" />
   </ItemGroup>
 
+  <ItemGroup>
+    <Folder Include="Persistence\Migrations\" />
+  </ItemGroup>
+
```
Comment (asynts) [inline src/Infrastructure/Infrastructure.csproj:23]: Why is this needed? My understanding is that all `.cs` files are automatically included by default.

File: src/Domain/Common/AuditEntity.cs [hunk 4]
```diff
@@ -0,0 +1,14 @@
+﻿using System;
```
Comment (asynts) [inline src/Domain/Common/AuditEntity.cs:1]: Something seems to be incorrectly configured here. I get the following warnings when building the solution:

~~~none
Entities/TagHistory.cs(9,21): warning CS0108: 'TagHistory.HistoryId' hides inherited member 'AuditEntity<Tag>.HistoryId'. Use the new keyword if hiding was intended. [/home/me/dev/codidact/core/src/Domain/Domain.csproj]
Entities/TagHistory.cs(10,25): warning CS0108: 'TagHistory.HistoryActivityAt' hides inherited member 'AuditEntity<Tag>.HistoryActivityAt'. Use the new keyword if hiding was intended. [/home/me/dev/codidact/core/src/Domain/Domain.csproj]
Entities/TagHistory.cs(13,25): warning CS0108: 'TagHistory.CreatedAt' hides inherited member 'AuditableEntity.CreatedAt'. Use the new keyword if hiding was intended. [/home/me/dev/codidact/core/src/Domain/Domain.csproj]
Entities/TagHistory.cs(14,25): warning CS0108: 'TagHistory.LastModifiedAt' hides inherited member 'AuditableEntity.LastModifiedAt'. Use the new keyword if hiding was intended. [/home/me/dev/codidact/core/src/Domain/Domain.csproj]
Entities/TagHistory.cs(15,21): warning CS0108: 'TagHistory.CreatedByMemberId' hides inherited member 'AuditableEntity.CreatedByMemberId'. Use the new keyword if hiding was intended. [/home/me/dev/codidact/core/src/Domain/Domain.csproj]
Entities/TagHistory.cs(16,21): warning CS0108: 'TagHistory.LastModifiedByMemberId' hides inherited member 'AuditableEntity.LastModifiedByMemberId'. Use the new keyword if hiding was intended. [/home/me/dev/codidact/core/src/Domain/Domain.csproj]
Entities/CategoryHistory.cs(16,21): warning CS0108: 'CategoryHistory.HistoryId' hides inherited member 'AuditEntity<Category>.HistoryId'. Use the new keyword if hiding was intended. [/home/me/dev/codidact/core/src/Domain/Domain.csproj]
Entities/CategoryHistory.cs(17,25): warning CS0108: 'CategoryHistory.HistoryActivityAt' hides inherited member 'AuditEntity<Category>.HistoryActivityAt'. Use the new keyword if hiding was intended. [/home/me/dev/codidact/core/src/Domain/Domain.csproj]
Entities/CategoryHistory.cs(18,21): warning CS0108: 'CategoryHistory.HistoryActivityMemberId' hides inherited member 'AuditEntity<Category>.HistoryActivityMemberId'. Use the new keyword if hiding was intended. [/home/me/dev/codidact/core/src/Domain/Domain.csproj]
Entities/CategoryHistory.cs(19,31): warning CS0114: 'CategoryHistory.HistoryActivityMember' hides inherited member 'AuditEntity<Category>.HistoryActivityMember'. To make the current member override that implementation, add the override keyword. Otherwise add the new keyword. [/home/me/dev/codidact/core/src/Domain/Domain.csproj]
~~~

I do not fully understand what is going on here, but it seems something was not generated properly here.

File: src/Domain/Common/AuditableEntity.cs [hunk 5]
```diff
@@ -1,4 +1,5 @@
 ﻿using System;
```
Comment (asynts) [inline src/Domain/Common/AuditableEntity.cs:1]: We now have a `AuditEntity` and an `AuditableEntity`. Maybe `BaseEntity` would be a better name?

File: src/Domain/Entities/Category.cs [hunk 6]
```diff
@@ -0,0 +1,34 @@
+﻿using System;
```
Comment (asynts) [inline src/Domain/Entities/Category.cs:1]: Since these files are generated you should add the following to `.gitattributes`:

~~~none
/src/Domain/Entities/* linguist-generated
~~~

File: src/Infrastructure/Persistence/Configuration/CategoryConfiguration.cs [hunk 7]
```diff
@@ -0,0 +1,53 @@
+﻿using Codidact.Core.Domain.Entities;
```
Comment (asynts) [inline src/Infrastructure/Persistence/Configuration/CategoryConfiguration.cs:1]: Since these files are generated you should add the following to `.gitattributes`:

~~~none
/src/Infrastructure/Persistence/Configuration/* linguist-generated
~~~

**Review by asynts — COMMENTED** [review]
Refer to #49.

File: src/Infrastructure/Infrastructure.csproj (hunk 3 above)
Comment (misha130) [reply src/Infrastructure/Infrastructure.csproj:23]: Since the folder is currently empty, vs added that

File: src/Domain/Common/AuditableEntity.cs (hunk 5 above)
Comment (misha130) [reply src/Domain/Common/AuditableEntity.cs:1]: One for the real entities, one for the audit. Maybe its too confusing

File: src/Domain/Entities/Category.cs (hunk 6 above)
Comment (misha130) [reply src/Domain/Entities/Category.cs:1]: I heavily edited them after they were generated. I moved the config to individual files, cleaned redundancy, added base classes. 

Honestly I think Ill scrap this PR and just do only generated files and some SQL script

Comment (asynts) [reply src/Domain/Entities/Category.cs:1]: Oh, I was not aware that the files were edited.

File: src/Domain/Common/AuditEntity.cs (hunk 4 above)
Comment (misha130) [reply src/Domain/Common/AuditEntity.cs:1]: Well since it isn't generated obviously I just missed this.
There was also a typo in TagHistory with the column being `HistoryActivityId` and not `HistoryActivityMemberId`.
Many such things were found after editing the files.

File: src/Infrastructure/Persistence/ApplicationDbContext.cs (hunk 2 above)
Comment (misha130) [reply src/Infrastructure/Persistence/ApplicationDbContext.cs:103]: I didn't. The scaffold did. I have to google now what it is to figure it out if it should stay

File: src/Domain/Entities/Member.cs [hunk 8]
```diff
@@ -1,82 +1,169 @@
...
+            SocialMediaTypeLastModifiedByMember = new HashSet<SocialMediaType>();
+            TagCreatedByMember = new HashSet<Tag>();
+            TagHistory = new HashSet<TagHistory>();
+            TagLastModifiedByMember = new HashSet<Tag>();
+            TrustLevelCreatedByMember = new HashSet<TrustLevel>();
+            TrustLevelHistory = new HashSet<TrustLevelHistory>();
+            TrustLevelLastModifiedByMember = new HashSet<TrustLevel>();
+            VoteTypeCreatedByMember = new HashSet<VoteType>();
+            VoteTypeHistory = new HashSet<VoteTypeHistory>();
+            VoteTypeLastModifeidByMember = new HashSet<VoteType>();
```
Comment (ArtOfCode-) [inline src/Domain/Entities/Member.cs]: Typo: `Modifeid` => `Modified`

**Review by ArtOfCode- — CHANGES_REQUESTED** [review]
Looks good. One typo, then good to merge AFAIC.

File: src/Domain/Entities/Member.cs (hunk 8 above)
Comment (misha130) [reply src/Domain/Entities/Member.cs]: https://github.com/codidact/core/pull/71/commits/204b3e847a38d1abdca70a6d4a52bd182074d146

**Review by ArtOfCode- — APPROVED** [review]
